packages/components/src/components/Button/Button.tsx

```diff
@@ -78,31 +86,27 @@ export const Button = forwardRef<HTMLButtonElement | null, ButtonProps>(
             <button
                 ref={ref}
                 data-test-id={dataTestId}
-                className={buttonStyles({
-                    iconOnly: (icon && !children) || hideLabel,
-                    size,
-                    style,
-                    ...props,
-                })}
+                className={cn(
+                    buttonStyles({
+                        size,
+                        style,
+                        ...props,
+                    }),
+                    textStyles({ style, ...props }),
+                    iconStyles({
+                        style,
+                        ...props,
+                    }),
+                    className,
+                )}
                 {...props}
             >
                 {icon && (
-                    <span
-                        data-test-id={`${dataTestId}-icon`}
-                        className={iconStyles({
-                            iconSpacing: children && !hideLabel ? size : 'none',
-                            style,
-                            ...props,
-                        })}
-                    >
+                    <span data-test-id={`${dataTestId}-icon`}>
                         {cloneElement(icon, { size: buttonIconSizeMap[size] })}
                     </span>
                 )}
```

Comment:
Same as the prop `icon`, if we agree to remove it, then should be removed too

Reply:
removed